### PR TITLE
minor tweak to pow accuracy and tests

### PR DESCRIPTION
**Pull request by oscardssmith**
This comes at a minor speed decrease (around 0.4 ns) but brings the accuracy for normal outputs to .53 ULP and .77 ULP for subnormal outputs (from roughly 1.01 before).

### CODE REVIEW

**Comment by giordano** [comment]
Is it worth adding explicit tests for pathological couples like `x, y = (0.0013653274095082324, -97.60372292227069)` from #48211, instead of waiting to randomly hitting them?

**Comment by oscardssmith** [comment]
Possibly, Other candidates would be (0.00186967536390803, 94.02052024653833) and (8.758520413376658e-5, 70.55863059215994)

**Comment by mikmoore** [comment]
I see that this proposal uses an explicit `fma`. This means that the (limited) set of users without native `fma` will see a major performance regression from this implementation.

**Comment by oscardssmith** [comment]
yeah. Without the `fma` the accuracy would be a bunch worse.

File: test/math.jl [hunk 1]
```diff
@@ -1396,6 +1398,9 @@ end
     # test for large negative exponent where error compensation matters
     @test 0.9999999955206014^-1.0e8 == 1.565084574870928
     @test 3e18^20 == Inf
+    # two cases where we have observed > 1 ULP in the past
+    @test 0.0013653274095082324^-97.60372292227069 == 4.088393948750035e279
+    @test 8.758520413376658e-5^70.55863059215994 == 5.052076767078296e-287
```
Comment (giordano) [inline test/math.jl:1403]: Thanks for adding the tests!  Should be guarded by `has_fma[Float64]`, or we'd get the same exact result also on non-FMA CPUs?

Comment (oscardssmith) [reply test/math.jl:1403]: It should, but I haven't tested (don't have especially easy access to a computer without).

Comment (gbaraldi) [reply test/math.jl:1403]: Can't you start julia with a specific flag? like --target=ivy-bridge ?

Comment (mikmoore) [reply test/math.jl:1403]: `fma` is emulated when not available directly from hardware. See `Base.fma_emulated` in `floatfuncs.jl`. The emulated version should be accurate, it's just very very slow.

Comment (mikmoore) [reply test/math.jl:1403]: If `fma` is important to accuracy, do not use `muladd`. `muladd` does not consistently convert to `fma` (even when it's natively supported). `muladd` gives the option to the compiler, but I find that whether it actually uses `fma` is largely a coin toss. It's also subject to change with inlining, optimization level, and architecture.

In particular, any test that a function produces exact values while using `muladd` seems prone to unpredictable failures.

Comment (oscardssmith) [reply test/math.jl:1403]: I've fairly carefully validated that the `muladd`s here do good things on computers with `fma`. It helps that most of these `muladds` work either way. They are designed such that all of them (other than the ones on lines 212 and 213) won't significantly effect the results.

Comment (oscardssmith) [reply test/math.jl:1403]: can confirm that both of these tests pass without fma. (which is not that surprising)

**Comment by oscardssmith** [comment]
should we backport this to 1.9?

**Pull request merged**